src/raw.rs

```diff
@@ -270,15 +318,15 @@ impl<'a> Future for Scan<'a> {
 
 pub struct BatchScan<'a> {
     client: &'a Client,
-    ranges: Vec<(Key, Key)>,
+    ranges: Vec<(Bound<Key>, Bound<Key>)>,
```

Comment:
What's your concern about collecting the iterator into a Vec? Is there any drawback of storing an iterator here (like the previous version)?

Reply:
I'd like to revisit doing that :)